### PR TITLE
More cleanup to annotations tutorial.

### PR DESCRIPTION
**Pull request by anntzer**
Make most of the tutorial a comment, rather than a string.  Remove
non-informative plot titles (as in #20393).

Move annotate_simple01 into the tutorial, and delete it as independent
example.

Remove the table of fancybox properties, and directly output them as
part of the plot in fancybox_demo.py.  In that same example, it is
also nicer to use two column output, and simpler to just output one
fancybox per subplot, rather than having to recompute the position of
each fancybox in data coordinates.

Don't alphabetically sort boxstyles/arrowstyles/etc., the source order
being semantically more meaningful (as in #20430).

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: examples/shapes_and_collections/fancybox_demo.py [hunk 1]
```diff
@@ -15,17 +17,16 @@
 # First we'll show some sample boxes with fancybox.
```
Comment (timhoffm) [inline examples/shapes_and_collections/fancybox_demo.py:17]:
```suggestion
# The following box styles are available:
```

File: examples/shapes_and_collections/fancybox_demo.py [hunk 2]
```diff
@@ -15,17 +17,16 @@
...
-             size=fontsize,
-             bbox=dict(boxstyle=stylename, fc="w", ec="k"))
+ncol = 2
+nrow = (len(styles) + 1) // ncol
+axs = plt.figure(figsize=(3 * ncol, nrow)).subplots(nrow, ncol)
+for ax in axs.flat:
+    ax.set_axis_off()
+for ax, (stylename, stylecls) in zip(axs.T.flat, styles.items()):
+    ax.text(.5, .5,
+            f"{stylename}\n{inspect.signature(stylecls)}",
```
Comment (timhoffm) [inline examples/shapes_and_collections/fancybox_demo.py:27]: I'm not convinced this stylecls inspection is an improvement. This plot should primarily illustrate the available boxstyles. Adding the parameters is rather distracting. And also these are only the defaults, which should be irrelevant to most users (well maybe if you start modifying them, but that should be discussed separately.

**Review by timhoffm — COMMENTED** [review]
Deleting annotate_simple01 seems somewhat arbitrary. annotate_simple01...04 sort of belong together. They show some examples of annotations. Whether we want them as separate examples (because the show up in the gallery) or whether that's too much can be discussed. But removing only 01 is inconsistent. Either we keep all, or we remove all.

My first-step intuition is to merge annotate_simple01..04 into a single example with a (2, 2) subplot. That way it's still possible to see the variants in the gallery, but they can be described and compared more easily within a single example.

**Comment by anntzer** [comment]
Sorry, I think I'm trying to do too much here, let me split that PR into smaller ones...

**Comment by anntzer** [comment]
Feel free to pick up whatever is relevant from my patch and either close this if you think everything that matters has been handled, or leave it open if there's more to do.

**Comment by QuLogic** [comment]
@story645 did you incorporate any of this PR? Or should we close this PR as outdated?

**Comment by anntzer** [comment]
OK, I think the only remaining work is #24723, everything else has been handled.

**Pull request closed without merging**